Superposition/ReferenceImplementation.qs

```diff
@@ -178,7 +178,46 @@ namespace Quantum.Kata.Superposition {
     }
     
     
-    // Task 10. Superposition of |0...0⟩ and given bit string
+    // ------------------------------------------------------
+    // Task 10. |00⟩ + |01⟩ + |10⟩ state
+    // Input: 2 qubits in |00⟩ state.
+    // Goal: create the state (|00⟩ + |01⟩ + |10⟩) / sqrt(3) on these qubits.
+    operation ThreeStates_TwoQubits_Reference (qs : Qubit[]) : Unit {
+        
+        body (...) {
+            // Follow Niel's answer at https://quantumcomputing.stackexchange.com/a/2313/
+            
+            // Rotate first qubit to (sqrt(2) |0⟩ + |1⟩) / sqrt(3) (task 1.4 from BasicGates kata)
+            let theta = ArcSin(1.0 / Sqrt(3.0));
+            Ry(2.0 * theta, qs[0]);
+            
+            // Split the state sqrt(2) |0⟩ ⊗ |0⟩ into |00⟩ + |01⟩
+            (ControlledOnInt(0, H))([qs[0]], qs[1]);
+        }
+        
+        adjoint auto;
+    }
+
+    // Alternative solution, based on post-selection
+    operation ThreeStates_TwoQubits_Postselection (qs : Qubit[]) : Unit {
+        using (ancilla = Qubit()) {
```

Comment:
In general, I suggest "auxillary" in preference to "ancilla."

Reply:
We use both terms interchangeably across our code base, so it's fine.